### DISCUSSION TITLE
Ant BMS connection to Goodwe inverter.

### DISCUSSION THREAD

**Discussion by Rainha1978** [syssi/esphome-ant-bms#47] [Q&A]
Hello.
It's possible to connect Ant BMS to Goodwee inverter with an ESP?

I have a battery with an ANT BMS connected by + and - wires to inverter, don't work in best conditions.
If it's possible, what it's need to put it working?
Inverter is Goodwe GW5048-ES.

Don't have much knowledge about ESP.

Thanks

Nuno Rainha

**Comment by syssi** [comment]
Thanks for creating this issue! :-) The component retrieves the BMS status (various values) periodically. These values can be exposed as sensor states. If you find a second component supporting your inverter you could forward the values to the inverter. There is no plug'n'play solution available for your usecase.

**Comment by Rainha1978** [comment]
Thanks for responding.

Soo, can't connect inverter by CAN or UART to the ESP to read those values? 
I think the most important value that inverter need is the SOC, because almost all program of the inverter works with that value, and it's the only problem I have this moment, the SOC isn't correct l.
The cycle of charge and discharge, only work's when battery is at 100%, after that the value varieties, and charging/discharging increase that difference.

**Comment by syssi** [comment]
This component enable you to retrieve the SoC value from the BMS. You need another solution / protocol implementation to provide the SoC value to your inverter. I started something like this here: 

https://github.com/syssi/esphome-virtual-can-bms

But the CAN support of ESPHome isn't stable at the moment and the implementation isn't complete because of the small user base / lack of testing.

**Comment by Rainha1978** [comment]
Ok.

I leave here some informations about configurations on the application.
Only to know...
![Screenshot_2023-01-23-14-14-57-537_com goodweforphone](https://user-images.githubusercontent.com/123375667/214062439-79ec96fc-9121-494c-bcfc-d1da7138038c.jpg)
![Screenshot_2023-01-23-14-15-49-689_com goodweforphone](https://user-images.githubusercontent.com/123375667/214062444-af31f4c5-f2d0-40e5-aa31-62cc9718ee9e.jpg)
![Screenshot_2023-01-23-14-16-56-315_com goodweforphone](https://user-images.githubusercontent.com/123375667/214062445-0a1242ce-2f9a-4b47-b18d-716665861f54.jpg)
